src/github.ts

```diff
@@ -466,6 +497,77 @@ export class GitHub {
     return refResponse.data.object.sha;
   }
 
+  /**
+   * Find the most "last" merged PR given a headBranch. "last" here means
+   * the most recently created. Includes all associated files.
+   *
+   * @param {string} headBranch - e.g. "release-please/branches/main"
+   * @returns {MergedGitHubPRWithFiles} - if found, otherwise undefined.
+   */
+  async lastMergedPRByHeadBranch(
+    headBranch: string
+  ): Promise<MergedGitHubPRWithFiles | undefined> {
+    const baseBranch = await this.getDefaultBranch();
+    const response: Repository<PullRequests> = await this.graphqlRequest({
+      query: `query lastMergedPRByHeadBranch($owner: String!, $repo: String!, $baseBranch: String!, $headBranch: String!) {
```

Comment:
I wish we weren't continuing to collect new GraphQL calls, but this looks solid to me.

Reply:
I could have gone this route in the manifest using existing GitHub functionality:
```typescript
const defaultBranch = await this.gh.getDefaultBranch();
const branchName = (await this.getBranchName()).toString()
const filter: MergedPullRequestFilter = (pr: MergedGitHubPR) => {
  return pr.baseRefName === defaultBranch && pr.headRefName === branchName;
}
const pr = this.gh.findMergedPullRequest(targetBranch, filter) // pages through `findMergedPullRequests()` results
const prWithFiles = this.gh.pullRequestFiles(pr.number);  // in a while loop to make sure we get all the files
```

but it seemed a little clunky when:
1/ given a stable branch name we can filter down to exactly the PR we want in the API call rather than in memory
2/ get everything in one* API call as opposed to a REST call and then a GraphQL call (*well, more than one if there are more than 100 files associated with the PR but that's true either way)

we should definitely take stock at some point of all the ways release-please needs to query the GitHub API to see if we can consolidate.

Reply:
> but it seemed a little clunky

I'm sold on the approach you've used.